asset-transfer-private-data/chaincode-go/README.md

```diff
@@ -0,0 +1,300 @@
+```
+
+Run the command below to copy the Node OU configuration file into the owner identity MSP folder.
+```
+cp ${PWD}/organizations/peerOrganizations/org1.example.com/msp/config.yaml ${PWD}/organizations/peerOrganizations/org1.example.com/users/owner@org1.example.com/msp/config.yaml
+```
+
+We can now use the Org2 CA to create the buyer identity. Set the Fabric CA client home the Org2 CA admin:
+```
+export FABRIC_CA_CLIENT_HOME=${PWD}/organizations/peerOrganizations/org2.example.com/
+```
+
+You can register a new owner client identity using the `fabric-ca-client` tool:
+```
+fabric-ca-client register --caname ca-org2 --id.name buyer --id.secret buyerpw --id.type client --tls.certfiles ${PWD}/organizations/fabric-ca/org2/tls-cert.pem
+```
+
+We can now enroll to generate the identity MSP folder:
+```
+fabric-ca-client enroll -u https://buyer:buyerpw@localhost:8054 --caname ca-org2 -M ${PWD}/organizations/peerOrganizations/org2.example.com/users/buyer@org2.example.com/msp --tls.certfiles ${PWD}/organizations/fabric-ca/org2/tls-cert.pem
+```
+
+Run the command below to copy the Node OU configuration file into the buyer identity MSP folder.
+```
+cp ${PWD}/organizations/peerOrganizations/org2.example.com/msp/config.yaml ${PWD}/organizations/peerOrganizations/org2.example.com/users/buyer@org2.example.com/msp/config.yaml
+```
+
+## Create an asset
+
+Now that we have created the identity of the asset owner, we can invoke the smart contract to create a new asset. Use the following environment variables to operate the `peer` CLI as the owner identity from Org1.
+
+```
+export CORE_PEER_LOCALMSPID="Org1MSP"
+export CORE_PEER_MSPCONFIGPATH=${PWD}/organizations/peerOrganizations/org1.example.com/users/owner@org1.example.com/msp
+export CORE_PEER_TLS_ROOTCERT_FILE=${PWD}/organizations/peerOrganizations/org1.example.com/peers/peer0.org1.example.com/tls/ca.crt
+export CORE_PEER_ADDRESS=localhost:7051
+```
+
+Run the following command to define the asset properties:
+```
+export ASSET_PROPERTIES=$(echo -n "{\"type\":\"asset\",\"asset_id\":\"asset1\",\"color\":\"green\",\"size\":20,\"appraised_value\":100}" | base64 | tr -d \\n)
+```
+
+We can the invoke the smart contract to create the new asset:
+```
+peer chaincode invoke -o localhost:7050 --ordererTLSHostnameOverride orderer.example.com --tls --cafile ${PWD}/organizations/ordererOrganizations/example.com/orderers/orderer.example.com/msp/tlscacerts/tlsca.example.com-cert.pem -C mychannel -n private_transfer -c '{"function":"CreateAsset","Args":[]}' --transient "{\"asset_properties\":\"$ASSET_PROPERTIES\"}"
+```
+
+The command above uses the transient data flag, `--transient`, to provide the marble details to the transaction. Transient data is is only stored in a transient data store in the peers targeted by the transaction. Transient data is not part of the transaction read write set, and as result is not stored on the channel ledger.
```

Comment:
```suggestion
The command above uses the transient data flag, `--transient`, to provide asset details to the chaincode. Transient data is not part of the transaction read write set, and as result is not stored on the channel ledger.
```

Reply:
Nik, there are two different 'transient' concepts that you were mixing up...
Transient field is used to pass private data input to chaincode on the endorsing peers, it is never stored.
Transient store is used to store private proposed writes locally so that they are available at commit time. All endorsers store their private writes, and also disseminate to other peers who also store the private writes until commit time.
We focus on the former here.

Reply:
Misread the documentation. My mistake.

Fixed.